src/components/app.tsx

```diff
@@ -1,21 +1,64 @@
-import React, { memo } from 'react';
-import { Box, color, ColorModeProvider, Flex, ThemeProvider } from '@stacks/ui';
+import React, { memo, useEffect } from 'react';
+import { useAtom } from 'jotai';
+import { color, ColorModeProvider, Flex, ThemeProvider } from '@stacks/ui';
+import { Connect } from '@stacks/connect-react';
+import { AuthOptions } from '@stacks/connect';
+import { appPrivateKeyAtom, authResponseAtom, userAtom, userSession } from '@store/auth';
 import { Header } from '@components/header';
 import { WelcomePanel } from '@components/welcome-panel';
 import 'modern-normalize/modern-normalize.css';
 
 export const App: React.FC = memo(() => {
+  const [appPrivateKey, setAppPrivateKey] = useAtom(appPrivateKeyAtom);
+  const [authResponse, setAuthResponse] = useAtom(authResponseAtom);
+  const [user, setUser] = useAtom(userAtom);
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      void setUser(user);
+    }
+  }, []);
```

Comment:
This seems unneeded. You're setting setUser with user.

you can also do this:
```ts
const isUserSignedInAtom = atom((get) => {
const userSession = get(userSessionAtom); // would need to make this
return userSession.isUserSignedIn();
});
```

but better yet, you'd just have something that is like this:

const userAtom = atomWithDefault((get) => { 
const userSession = get(userSessionAtom);
return userSession.loadUserData();
});
```

Reply:
Yeah, good point. I need to look it over a bit bc I was referencing the current test-app auth and there seems to be some code there that needs to be refactored too. I had stuff moved out in a `use-auth` hook but pulled it back in bc thought seemed unneeded. Let me refactor with your suggestion.